docker/bootstrap/Dockerfile.common

```diff
@@ -58,13 +56,6 @@ COPY config /vt/src/vitess.io/vitess/config
 COPY third_party /vt/src/vitess.io/vitess/third_party
 COPY tools /vt/src/vitess.io/vitess/tools
 COPY travis /vt/src/vitess.io/vitess/travis
-COPY vendor/vendor.json /vt/src/vitess.io/vitess/vendor/
-
-# Download vendored Go dependencies
-RUN cd /vt/src/vitess.io/vitess && \
```

Comment:
An observation: without a `go mod download` here, the cost of building base images from bootstrap images would be substantially increased, since each base image create needs to re-download all the module deps.

The downloads will be repeated work for each base image.

The downside of running `go mod download` here is that it'll make the bootstrap image even larger than it was before with `govendor sync` because modules are not yet as good about omitting unnecessary deps. It's possible that setting `GOPROXY=https://goproxy.io` and/or upgrading to go 1.13 where that's the default could help with module downloading speed and size regardless, but that official proxy super new and I haven't tested it yet to see if download sizes are smaller vs fetching whole git repos.

I'm not necessarily advocating for or against putting `go mod download` here, but it seems worth consideration (and setting GOPROXY=https://goproxy.io` seems worth consideration also, as long as we're on 1.12)

Reply:
I have added the `go mod download` here - I think it makes sense. Let's reevaluate GOPROXY after the 1.13 bump. We can guarantee 1.12, so I'm open to doing it earlier if we need to.